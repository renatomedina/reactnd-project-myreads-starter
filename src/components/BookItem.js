import React from 'react';
import PropTypes from 'prop-types'

const BookItem =  ({ title = "", author = "", backgroundImage, onChangeShelf, shelf = "none" }) => {

    return (
        <div className="book">
            <div className="book-top">
                <div className="book-cover" style={{ width: 128, height: 193, backgroundImage: `url(${backgroundImage})` }}></div>
                <div className="book-shelf-changer">
                    <select value={shelf} onChange={e => { onChangeShelf(e) }}>
                        <option value="none" disabled>Move to...</option>
                        <option value="currentlyReading">Currently Reading</option>
                        <option value="wantToRead">Want to Read</option>
                        <option value="read">Read</option>
                        <option value="none">None</option>
                    </select>
                </div>
            </div>
            <div className="book-title">{title}</div>
            <div className="book-authors">{author}</div>
        </div>
    );
}

BookItem.propTypes = {
    backgroundImage: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
    author: PropTypes.arrayOf(PropTypes.string).isRequired,
    shelf: PropTypes.string.isRequired,
    onChangeShelf: PropTypes.func.isRequired,
  }

  export default BookItem;