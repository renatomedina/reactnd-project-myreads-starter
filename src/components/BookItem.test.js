import React from 'react';
import { shallow } from 'enzyme';
import BookItem from './BookItem';

describe('<BookItem />', () => {
    it('shallow renders correctly', () => {
        expect(shallow(<BookItem />))
    })

    it('expects render 5 options in select', () => {
        const wrapper = shallow(<BookItem />)
        expect(wrapper.find('option').length).toBe(5)
    })

    it('expects onChangeShelf to be colled on shelf', () => {
        const onChangeShelf = jest.fn()
        const wrapper = shallow(<BookItem onChangeShelf={onChangeShelf} />)
        wrapper.find('select').simulate('change')
        expect(onChangeShelf).toHaveBeenCalledTimes(1)
    })
}) 