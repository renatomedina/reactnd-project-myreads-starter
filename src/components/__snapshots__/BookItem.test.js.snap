// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BookItem /> shallow renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <BookItem
    author={
      Array [
        "Marcio G.",
      ]
    }
    backgroundImage="https://i.ytimg.com/vi/PKffm2uI4dk/maxresdefault.jpg"
    onChangeShelf={[Function]}
    shelf="read"
    title="React Development"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="book-top"
        >
          <div
            className="book-cover"
            style={
              Object {
                "backgroundImage": "url(https://i.ytimg.com/vi/PKffm2uI4dk/maxresdefault.jpg)",
                "height": 193,
                "width": 128,
              }
            }
          />
          <div
            className="book-shelf-changer"
          >
            <select
              onChange={[Function]}
              value="read"
            >
              <option
                disabled={true}
                value="none"
              >
                Move to...
              </option>
              <option
                value="currentlyReading"
              >
                Currently Reading
              </option>
              <option
                value="wantToRead"
              >
                Want to Read
              </option>
              <option
                value="read"
              >
                Read
              </option>
              <option
                value="none"
              >
                None
              </option>
            </select>
          </div>
        </div>,
        <div
          className="book-title"
        >
          React Development
        </div>,
        <div
          className="book-authors"
        >
          Marcio G.
        </div>,
      ],
      "className": "book",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="book-cover"
              style={
                Object {
                  "backgroundImage": "url(https://i.ytimg.com/vi/PKffm2uI4dk/maxresdefault.jpg)",
                  "height": 193,
                  "width": 128,
                }
              }
            />,
            <div
              className="book-shelf-changer"
            >
              <select
                onChange={[Function]}
                value="read"
              >
                <option
                  disabled={true}
                  value="none"
                >
                  Move to...
                </option>
                <option
                  value="currentlyReading"
                >
                  Currently Reading
                </option>
                <option
                  value="wantToRead"
                >
                  Want to Read
                </option>
                <option
                  value="read"
                >
                  Read
                </option>
                <option
                  value="none"
                >
                  None
                </option>
              </select>
            </div>,
          ],
          "className": "book-top",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "book-cover",
              "style": Object {
                "backgroundImage": "url(https://i.ytimg.com/vi/PKffm2uI4dk/maxresdefault.jpg)",
                "height": 193,
                "width": 128,
              },
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <select
                onChange={[Function]}
                value="read"
              >
                <option
                  disabled={true}
                  value="none"
                >
                  Move to...
                </option>
                <option
                  value="currentlyReading"
                >
                  Currently Reading
                </option>
                <option
                  value="wantToRead"
                >
                  Want to Read
                </option>
                <option
                  value="read"
                >
                  Read
                </option>
                <option
                  value="none"
                >
                  None
                </option>
              </select>,
              "className": "book-shelf-changer",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <option
                    disabled={true}
                    value="none"
                  >
                    Move to...
                  </option>,
                  <option
                    value="currentlyReading"
                  >
                    Currently Reading
                  </option>,
                  <option
                    value="wantToRead"
                  >
                    Want to Read
                  </option>,
                  <option
                    value="read"
                  >
                    Read
                  </option>,
                  <option
                    value="none"
                  >
                    None
                  </option>,
                ],
                "onChange": [Function],
                "value": "read",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Move to...",
                    "disabled": true,
                    "value": "none",
                  },
                  "ref": null,
                  "rendered": "Move to...",
                  "type": "option",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Currently Reading",
                    "value": "currentlyReading",
                  },
                  "ref": null,
                  "rendered": "Currently Reading",
                  "type": "option",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Want to Read",
                    "value": "wantToRead",
                  },
                  "ref": null,
                  "rendered": "Want to Read",
                  "type": "option",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Read",
                    "value": "read",
                  },
                  "ref": null,
                  "rendered": "Read",
                  "type": "option",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "None",
                    "value": "none",
                  },
                  "ref": null,
                  "rendered": "None",
                  "type": "option",
                },
              ],
              "type": "select",
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "React Development",
          "className": "book-title",
        },
        "ref": null,
        "rendered": "React Development",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Marcio G.",
          ],
          "className": "book-authors",
        },
        "ref": null,
        "rendered": Array [
          "Marcio G.",
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="book-top"
          >
            <div
              className="book-cover"
              style={
                Object {
                  "backgroundImage": "url(https://i.ytimg.com/vi/PKffm2uI4dk/maxresdefault.jpg)",
                  "height": 193,
                  "width": 128,
                }
              }
            />
            <div
              className="book-shelf-changer"
            >
              <select
                onChange={[Function]}
                value="read"
              >
                <option
                  disabled={true}
                  value="none"
                >
                  Move to...
                </option>
                <option
                  value="currentlyReading"
                >
                  Currently Reading
                </option>
                <option
                  value="wantToRead"
                >
                  Want to Read
                </option>
                <option
                  value="read"
                >
                  Read
                </option>
                <option
                  value="none"
                >
                  None
                </option>
              </select>
            </div>
          </div>,
          <div
            className="book-title"
          >
            React Development
          </div>,
          <div
            className="book-authors"
          >
            Marcio G.
          </div>,
        ],
        "className": "book",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="book-cover"
                style={
                  Object {
                    "backgroundImage": "url(https://i.ytimg.com/vi/PKffm2uI4dk/maxresdefault.jpg)",
                    "height": 193,
                    "width": 128,
                  }
                }
              />,
              <div
                className="book-shelf-changer"
              >
                <select
                  onChange={[Function]}
                  value="read"
                >
                  <option
                    disabled={true}
                    value="none"
                  >
                    Move to...
                  </option>
                  <option
                    value="currentlyReading"
                  >
                    Currently Reading
                  </option>
                  <option
                    value="wantToRead"
                  >
                    Want to Read
                  </option>
                  <option
                    value="read"
                  >
                    Read
                  </option>
                  <option
                    value="none"
                  >
                    None
                  </option>
                </select>
              </div>,
            ],
            "className": "book-top",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "book-cover",
                "style": Object {
                  "backgroundImage": "url(https://i.ytimg.com/vi/PKffm2uI4dk/maxresdefault.jpg)",
                  "height": 193,
                  "width": 128,
                },
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <select
                  onChange={[Function]}
                  value="read"
                >
                  <option
                    disabled={true}
                    value="none"
                  >
                    Move to...
                  </option>
                  <option
                    value="currentlyReading"
                  >
                    Currently Reading
                  </option>
                  <option
                    value="wantToRead"
                  >
                    Want to Read
                  </option>
                  <option
                    value="read"
                  >
                    Read
                  </option>
                  <option
                    value="none"
                  >
                    None
                  </option>
                </select>,
                "className": "book-shelf-changer",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <option
                      disabled={true}
                      value="none"
                    >
                      Move to...
                    </option>,
                    <option
                      value="currentlyReading"
                    >
                      Currently Reading
                    </option>,
                    <option
                      value="wantToRead"
                    >
                      Want to Read
                    </option>,
                    <option
                      value="read"
                    >
                      Read
                    </option>,
                    <option
                      value="none"
                    >
                      None
                    </option>,
                  ],
                  "onChange": [Function],
                  "value": "read",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Move to...",
                      "disabled": true,
                      "value": "none",
                    },
                    "ref": null,
                    "rendered": "Move to...",
                    "type": "option",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Currently Reading",
                      "value": "currentlyReading",
                    },
                    "ref": null,
                    "rendered": "Currently Reading",
                    "type": "option",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Want to Read",
                      "value": "wantToRead",
                    },
                    "ref": null,
                    "rendered": "Want to Read",
                    "type": "option",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Read",
                      "value": "read",
                    },
                    "ref": null,
                    "rendered": "Read",
                    "type": "option",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "None",
                      "value": "none",
                    },
                    "ref": null,
                    "rendered": "None",
                    "type": "option",
                  },
                ],
                "type": "select",
              },
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "React Development",
            "className": "book-title",
          },
          "ref": null,
          "rendered": "React Development",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Marcio G.",
            ],
            "className": "book-authors",
          },
          "ref": null,
          "rendered": Array [
            "Marcio G.",
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
